chain/chain/src/resharding/event_type.rs

```diff
@@ -0,0 +1,184 @@
+//! Collection of all resharding V3 event types.
+
+use near_chain_primitives::Error;
+use near_primitives::hash::CryptoHash;
+use near_primitives::shard_layout::ShardLayout;
+use near_primitives::types::AccountId;
+use near_store::ShardUId;
+use tracing::error;
+
+/// Struct used to destructure a new shard layout definition into the resulting resharding event.
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub enum ReshardingEventType {
+    /// Split of a shard.
+    SplitShard(ReshardingSplitShardParams),
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub struct ReshardingSplitShardParams {
+    // Shard being split.
+    pub parent_shard: ShardUId,
+    // Child to the left of the account boundary.
+    pub left_child_shard: ShardUId,
+    // Child to the right of the account boundary.
+    pub right_child_shard: ShardUId,
+    /// The account at the boundary between the two children.
+    pub boundary_account: AccountId,
+    /// Hash of the first block having the new shard layout.
+    pub block_hash: CryptoHash,
```

Comment:
The hash of the first block with new shard layout is not known at the time when the flat storage resharding is triggered (unless maybe I'm missing something). My thinking was that this will be triggered together with all of the rest of resharding logic in post process of the last block of old shard layout.